save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/UsersDetailsController.kt

```diff
@@ -36,18 +38,22 @@ class UsersDetailsController(
 
     /**
      * @param newUserInfo
+     * @param authentication an [Authentication] representing an authenticated request
      */
     @PostMapping("/save")
     @PreAuthorize("isAuthenticated()")
-    fun saveUser(@RequestBody newUserInfo: UserInfo) {
+    fun saveUser(@RequestBody newUserInfo: UserInfo, authentication: Authentication) {
         val user = userRepository.findByName(newUserInfo.name).get()
-        userRepository.save(user.apply {
-            email = newUserInfo.email
-            company = newUserInfo.company
-            company = newUserInfo.location
-            company = newUserInfo.gitHub
-            company = newUserInfo.linkedin
-            company = newUserInfo.twitter
-        })
+        val userId = (authentication.details as AuthenticationDetails).id
+        if (user.id == userId) {
+            userRepository.save(user.apply {
+                email = newUserInfo.email
+                company = newUserInfo.company
+                company = newUserInfo.location
+                company = newUserInfo.gitHub
+                company = newUserInfo.linkedin
+                company = newUserInfo.twitter
+            })
+        }
```

Comment:
`else throw ResponseStatusException(HttpStatus.FORBIDDEN)`
Now you are not doing anything, but still returning `200 OK`

Reply:
fixed